src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionSettings.cs

```diff
@@ -51,7 +51,7 @@ internal sealed class HttpConnectionSettings
         public HttpConnectionSettings()
         {
             _maxHttpVersion = AllowHttp2 ? HttpVersion.Version20 : HttpVersion.Version11;
-            _allowPlainHttp2 = AllowPlainHttp2;
+            _allowUnencryptedHttp2 = AllowHttp2 && AllowUnencryptedHttp2;
```

Comment:
Nit: read AllowHttp2 into a local bool and use that bool twice rather than reading the property twice.  That's a) to avoid some minor cost associated with checking the same switches/envvars twice, and b) to avoid any weird race conditions where it switches between the above checks.